tests/testthat/test-MapFeatures.R

```diff
@@ -13,69 +13,218 @@ se_merged <- MergeSTData(se_mbrain, se_mcolon) |>
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), pt_alpha = 0.5), "patchwork")
   
-  # Test pt_stroke parameter
+  ## Test pt_stroke parameter
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), pt_stroke = 0.5), "patchwork")
   
-  # Test scale_alpha parameter
+  ## Test scale_alpha parameter
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), scale_alpha = TRUE), "patchwork")
   
-  # Test section_number parameter
+  ## Test section_number parameter
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), section_number = 2), "patchwork")
   
-  # Test label_by parameter
-  expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), label_by = "sample_id"), "patchwork")
+  ## Test label_by parameter
   p <- MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), label_by = "sample_id")
+  expect_s3_class(p, "patchwork")
   lbls <- c(p$labels$title, p$patches$plots[[1]]$labels$title)
   expect_true(all(lbls == c("mousecolon", "mousebrain")))
   
-  # Test ncol parameter
+  ## Test ncol parameter
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), ncol = 2), "patchwork")
   
-  # Test blend parameter
+  ## Test blend parameter
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), blend = TRUE), "patchwork")
   
-  # Test scalebar parameters
+  ## Test scalebar parameters
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), add_scalebar = TRUE), "patchwork")
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), add_scalebar = TRUE, scalebar_gg = scalebar(x = 1000)), "patchwork")
   
-  # Test colors parameter
+  ## Test colors parameter
   expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), colors = magma(n = 11, direction = -1)), "patchwork")
   
-  # Test that the correct samples and features are plotted
+  ## Test that the correct samples and features are plotted
   p <- MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), return_plot_list = TRUE)
   
-  # Test that the list contains correct classes
+  ## Test that the list contains correct classes
   expect_true(all(sapply(p, class) == "list"))
   expect_true(all(sapply(p[[1]], function(p) class(p)[1]) == "gg"))
   
-  # Test that the correct features are plotted
+  ## Test that the correct features are plotted
   expect_true(all(c("1", "2") == names(p)))
   expect_true(all(c("Clu", "Slc6a3", "Vip") == names(p[[1]])))
   
-  # Test that the expected number of plots are produced
+  ## Test that the expected number of plots are produced
   p <- MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"))
   expect_equal(p$patches$layout$ncol, 3)
+  
+  # Tile shape
+  ## Test image_use parameter
+  expect_s3_class(MapFeatures(se_merged, features = c("Clu", "Slc6a3", "Vip"), image_use = "raw", shape = "tile"), "patchwork")
```

Comment:
Preferably, the tests should make sure that the content of the plots is as expected. It is possible to extract parts of the patchwork object to verify that a parameter setting works as expected. Just checking the class is not going to be very helpful on its own.